mica/starcheck/starcheck.py

```diff
@@ -169,211 +172,157 @@ def get_starcheck_catalog_at_date(date, starcheck_db=None, timelines_db=None):
-        # should have been commanded.  If there are multiple dwells for attitude, try to
-        # use nman_start if available.
-        if dwell.manvr.n_dwell > 1 and dwell.manvr.nman_start is not None:
-            start_cat_search = dwell.manvr.nman_start
-        else:
-            start_cat_search = dwell.get_previous().stop
 
-    timelines = timelines_db.fetchall(
-        """select * from timeline_loads where scs < 131
-           and datestop > '{}' and datestart < '{}' order by datestart""".format(
-            start_cat_search, dwell_start
-        )
-    )
-    for timeline in timelines[::-1]:
-        starchecks = db.fetchall(
-            """select * from starcheck_obs, starcheck_id
-               where dir = '{}'
-               and mp_starcat_time >= '{}'
-               and mp_starcat_time <= '{}' and mp_starcat_time <= '{}'
-               and starcheck_id.id = starcheck_obs.sc_id
-               order by mp_starcat_time """.format(
-                timeline["mp_dir"],
-                timeline["datestart"],
-                timeline["datestop"],
-                dwell_start,
-            )
-        )
-        # The last one should be the one before beginning of the dwell
-        if len(starchecks):
-            # Use the obsid and the known products directory to use the more
-            # generic get_starcheck_catalog to fetch the right one from the
-            # database
-            cat_info = get_starcheck_catalog(
-                starchecks[-1]["obsid"], mp_dir=starchecks[-1]["dir"]
-            )
-            cat_info["status"] = "ran" if date < DateTime().date else "approved"
-            return cat_info
-    return None
+    # Get observations within +/- 7 days and find the one where ``date`` is between end
+    # of the previous obs dwell through the end of the current obs dwell.
+    obss = Table(get_observations(start=date - 7 * u.day, stop=date + 7 * u.day))
+    obss.sort("obs_start")
+    for obs_prev, obs in zip(obss[:-1], obss[1:]):
+        if obs_prev["obs_stop"] < date.date <= obs["obs_stop"]:
+            break
+    else:
+        raise ValueError(f"could not find observation at {date}")
 
+    mp_dir = load_name_to_mp_dir(obs["source"])
 
-def get_mp_dir(obsid, starcheck_db=None, timelines_db=None):
-    """
-    Get the mission planning directory for an obsid and some status information.
-    If the obsid catalog was used more than once (multi-obi or rescheduled after
-    being used in a vehicle-only interval), return the directory and details of
-    the last one used on the spacecraft.
+    # Kadi observations are labeled by the commanded obsid.  Starcheck catalogs
+    # are labeled by the planned obsid used in the starcheck output.  For observations
+    # after SCS107 with SOSA, the kadi observation obsids and starcheck obsids don't match.
+    # Use a helper function to line these up by starcat time.
+    obsid = get_starcheck_db_obsid(obs["starcat_date"], mp_dir)
```

Comment:
I've forgotten the best check for if an element is masked, but this would fix this one tiny additional use case.

```suggestion
    import numpy.ma as ma
    if ma.is_masked(obs['starcat_date']):
        obsid = obs["obsid"]
    else:
    	obsid = get_starcheck_db_obsid(obs["starcat_date"], mp_dir)```